metomi/isodatetime/data.py

```diff
@@ -302,33 +343,57 @@ def __iter__(self):
             else:
                 point = self.get_next(point)
 
+    def __hash__(self) -> int:
+        # TODO: fix when fixing issue #45
+        return hash((self._repetitions, self._start_point, self._end_point,
+                     self._min_point, self._max_point))
+
+    def __eq__(self, other: "TimeRecurrence") -> bool:
+        if not isinstance(other, TimeRecurrence):
+            return NotImplemented
+        # TODO: fix when fixing issue #45
+        for attr in ["repetitions", "start_point", "end_point", "min_point",
+                     "max_point"]:
+            if getattr(self, attr) != getattr(other, attr):
+                return False
+        return True
+
     def __str__(self):
-        if self.repetitions is None:
+        if self._repetitions is None:
             prefix = "R/"
         else:
-            prefix = "R" + str(self.repetitions) + "/"
-        if self.format_number == 1:
-            return prefix + str(self.start_point) + "/" + str(self.end_point)
-        elif self.format_number == 3:
-            return prefix + str(self.start_point) + "/" + str(self.duration)
-        elif self.format_number == 4:
-            return prefix + str(self.duration) + "/" + str(self.end_point)
+            prefix = "R" + str(self._repetitions) + "/"
+        if self._format_number == 1:
+            return prefix + str(self._start_point) + "/" + str(self._end_point)
+        elif self._format_number == 3:
+            return prefix + str(self._start_point) + "/" + str(self._duration)
+        elif self._format_number == 4:
+            return prefix + str(self._duration) + "/" + str(self._end_point)
         return "R/?/?"
 
+    def __repr__(self):
+        return "<isodatetime.data.TimeRecurrence: {0}>".format(repr(str(self)))
```

Comment:
Maybe we should use something like (not tested) `f"<{self.__class__.__module__}.{self.__class__.__name__}: {repr(str(self))}}>`, just in case it gets moved/renamed?

Reply:
Not needed but could iterate `self.__slots__` to make the `repr` reproducible e.g: `TimeRecurrence(years=1, months=1)`

Reply:
Unfortunately f strings aren't supported by Python 3.5